client/src/test/java/model/ShipCreatorTest.java

```diff
@@ -0,0 +1,45 @@
+package model;
+
+import gui.fields.Field;
+import org.testng.annotations.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+
+public class ShipCreatorTest {
+
+  @Test
+  public void whenShipIsNotFullyBuildMastIsAddedToShip() {
+
+    //given
+    int shipLength = 2;
+    Ship ship = new Ship(shipLength);
+    Field mast = mock(Field.class);
+
+    ShipCreator shipCreator = new ShipCreator(ship);
```

Comment:
you can extract to setUp method shipCreator, ship and mast - used in both test

Reply:
I cant do that, because shipCreator gets different ships in constructor